dune_client/api/custom.py

```diff
@@ -12,7 +12,7 @@
     ResultsResponse,
 )
 
-
+# pylint: disable=duplicate-code
```

Comment:
This is likely because of all the filter arguments and suggests a deeper change to the project. Introduce a `ResultFilter` structure that contains all the filtering possibilities:

```py
@dataclass
class ResultFilter:
    limit: Optional[int] = None
    offset: Optional[int] = None
    columns: Optional[List[str]] = None
    sample_count: Optional[int] = None
    filters: Optional[str] = None
    sort_by: Optional[List[str]] = None
```

Then here would look more like:
```py
    def get_custom_endpoint_result(
        self,
        handle: str,
        endpoint: str,
        filter: Optional[ResultFilter]
    ) -> ResultsResponse:
```


It would resolve both duplicated code lint warning and also reduce the "argument overload" lint error that is being globally suppressed here https://github.com/duneanalytics/dune-client/blob/ed3836d68522fabb421a663a881e4c3d98102370/.pylintrc#L4

Reply:
I will log this and do at another time.